specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json

```diff
@@ -0,0 +1,1671 @@
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ConfigurationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/singleServers/{serverName}/updateConfigurations": {
+      "post": {
+        "tags": [
+          "Configurations"
+        ],
+        "operationId": "Configurations_Update",
+        "x-ms-examples": {
+          "ConfigurationUpdate": {
+            "$ref": "./examples/ConfigurationUpdate.json"
+          }
+        },
+        "description": "Updates a configuration of a server.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ConfigurationUpdateList"
+            },
+            "description": "The required parameters for updating a server configuration."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ConfigurationListResult"
+            }
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/singleServers/{serverName}/maintenanceWindows/customerMaintenanceWindow": {
```

Comment:
>customerMaintenanceWindow [](start = 154, length = 25)

I think you're missing an API to list `maintenanceWindows`.

Also, will there be any other windows in the future or will it always be just `customerMaintenanceWindow`?

Reply:
we dont really support list for now, in the get call it always returns one record if there is any custom maintenance window otherwise it will return empty array. that's why it is having get call with return as array.

Reply:
This doesn't make any sense to me and is not an approved ARM pattern. The response body of GET, PUT, PATCH, DELETE on the same path MUST be a single resource payload - no arrays. You can have an array on a LIST call.

To support API exploration tools, you should have a list call like I said (yes, even if exactly 1 resource is always returned in the array). It shouldn't be a huge deal - you could even do it on the same controller.

---
In reply to: [451832510](https://github.com/Azure/azure-rest-api-specs/pull/10020#discussion_r451832510) [](ancestors = 451832510)

Reply:
Agreed. Updated the API.